pkg/cloud-controller-manager/loadBalancer.go

```diff
@@ -65,11 +78,11 @@ func (l *LoadBalancerManager) GetLoadBalancer(ctx context.Context, clusterName s
 }
 
 func (l *LoadBalancerManager) GetLoadBalancerName(ctx context.Context, clusterName string, service *v1.Service) string {
-	name := clusterName + "-" + service.Namespace + "-" + service.Name + "-"
+	name := clusterName + "-" + service.Namespace + "-" + service.Name
 
-	digest := crc32.ChecksumIEEE([]byte(name + string(service.UID)))
-	suffix := fmt.Sprintf("%08x", digest) // print in 8 width and pad with 0's
-	name += suffix
+	digest := crc32.ChecksumIEEE([]byte(name))
+	suffix := fmt.Sprintf("%08x", digest)
+	name = fmt.Sprintf("%s-%s", name, suffix)
```

Comment:
considering #L89, a better way is to trim name here to no more than 54 chars , then connect them, otherweise, it may still be duplicated, as digest may be stripped